libp2p/peerstore.nim

```diff
@@ -0,0 +1,140 @@
+## Nim-LibP2P
+## Copyright (c) 2021 Status Research & Development GmbH
+## Licensed under either of
+##  * Apache License, version 2.0, ([LICENSE-APACHE](LICENSE-APACHE))
+##  * MIT license ([LICENSE-MIT](LICENSE-MIT))
+## at your option.
+## This file may not be copied, modified, or distributed except according to
+## those terms.
+
+import
+  std/[tables, sets, sequtils],
+  ./peerid,
+  ./multiaddress
+
+type
+  ##############################
+  # Listener and handler types #
+  ##############################
+  
+  AddrChangeHandler* = proc(peerId: PeerID, multiaddrs: HashSet[MultiAddress])
```

Comment:
Is this communicating change per se (as in a delta) or is the set here the sum of all, existing and new, multiaddresses?

Reply:
Ah, yes, this could have been clearer. It's not merely the delta, but communicates all known multiaddrs per peer whenever there's a change as per [js-implementation](https://github.com/libp2p/js-libp2p/blob/master/doc/API.md#known-multiaddrs-for-a-peer-change).